record.c

```diff
@@ -384,17 +416,33 @@ int janus_recorder_save_frame(janus_recorder *recorder, char *buffer, uint lengt
 				res, sizeof(gint64), g_strerror(errno));
 		}
 	}
+	/* Edit packet header if needed */
+	rtp_header *header = (rtp_header *)buffer;
+	uint32_t ssrc = ntohl(header->ssrc);
+	uint16_t seq = ntohs(header->seq_number);
+	timestamp = ntohl(header->timestamp);
+	if(recorder->type != JANUS_RECORDER_DATA) {
```

Comment:
Accessing RTP fields should be done ONLY if it's not a data recorder, so this check needs to be moved up. The way it is, if you're recording a small data message (<12 bytes) this will crash.